sim/core/sim.go

```diff
@@ -253,7 +256,19 @@ func (sim *Simulation) RunOnce() {
 }
 
 func (sim *Simulation) AddPendingAction(pa *PendingAction) {
-	sim.pendingActions = append(sim.pendingActions, pa)
+	handled := false
```

Comment:
totally forgot to insert sorted. It just so happens that the dot tick time of moonfire is always longer than the next cast for a druid so we never hit this edge case.

Found it writing spriest code :)